Add semantics - more variety of elements and better hierchy of DOM.

andrewpking opened the pull request:
I added semantics as much as possible to breakup the layout of each webpage into sections corresponding to their roles and visual layout.

### Big Changes

- All pages now have a <main> tag.
- The sidebar now has an <aside> tag.
- The header now has a <header> tag.
- The footer now has a <footer> tag.

### Small Changes

- Removed unnecessary span from KBCS address and wrapped it in an <address> tag instead.
- Searchform.php has aria-roles merged into it from the home page.
- Closed a tag on one of the inputs on the home page and in searchform.php, it was cluttering my IDE with red.
- Moved <h2> into same div as <p> for homepage.
- Attempted to add <article> tags to blog posts on the main page, they are not rendering yet.

# TODOs:

- @andrewpking figure out how to make <article> tags show up in DOM.
- @taija review these changes.

andrewpking committed:
BC Logo no longer overlaps text on small screens.

andrewpking committed:
Styles created for donate buttons to be created on production site.

andrewpking committed:
Fix nav issues in header by adding nav tags to DOM.

andrewpking committed:
Added an aria label to search.

andrewpking committed:
Alt text for KBCS logo indicates that it links to homepage.

andrewpking committed:
Now using roles instead of ARIA labels for search.

andrewpking committed:
Search form now includes labels.

andrewpking committed:
Link to support us is more verbose.

andrewpking committed:
Added styles that make the touch targets larger.

andrewpking committed:
Made link for support shorter in footer.

andrewpking committed:
Added subclass to portion of footer with bclogo

andrewpking committed:
Fixed CSS issues caused by larger fonts in footer.

andrewpking committed:
Reflowed styles to be alphabetized in tablet media screen.

andrewpking committed:
header is now contained in a <header> tag.

andrewpking committed:
The body of the content is now in a <main> tag. Still working on getting blog posts to show up as <article>

andrewpking committed:
Added footer and address aria tags, got rid of the <span> for address.

andrewpking committed:
Added styles to street-address class in footer.

andrewpking committed:
Generic pages now are contained within a <main> tag.

andrewpking committed:
Tried adding a main to the search page, not having luck.

andrewpking committed:
Sidebar is now wrapped in an <aside> tag.

andrewpking committed:
Added <main> tag to search, it has its own page in the theme.

andrewpking committed:
Searchform.php  now has a role for search input.

andrewpking committed:
Added <main> tag to all relevant pages.

andrewpking committed:
Social links now have the role of button. They are now nested in a nav.

andrewpking committed:
Added the nav.

taija commented on footer.php:
```diff
@@ -2,29 +2,27 @@
 			<div class="row">
 
-				<div class="span3">
+				<nav class="span3">
```
If there is more than one `<nav>` element, each nav needs to be labeled with their purpose. You should either use `aria-label` to add a label directly, or use `aria-labelledby` to create a connection to the labelling element if one exists. See https://www.w3.org/WAI/tutorials/page-structure/labels/#using-aria-labelledby for details.

andrewpking replied:
Thanks for the resource! I updated the code to work with this.

taija commented on footer.php:
```diff
@@ -2,29 +2,27 @@
+				</nav><!-- span3 -->
                 
-                <div class="span3">
+                <nav class="span3">
```
I would recommend placing all of these in a single `<nav>` instead of having multiple small `<nav>` elements in the footer

taija replied:
Each of these navs also needs to be labelled (see previous comment)

taija commented on footer.php:
```diff
@@ -44,36 +42,36 @@
                 	
                 	<p class="bc-service">91.3 KBCS is a public service at</p>
-                    <a href="https://www.bellevuecollege.edu"><img src="<?php bloginfo('stylesheet_directory'); ?>/img/bellevuecollege.png" alt="Bellevue College" /></a></div> <!--bclogo-->
+                    <a role="button" href="https://www.bellevuecollege.edu"><img src="<?php bloginfo('stylesheet_directory'); ?>/img/bellevuecollege.png" alt="Bellevue College" /></a></div> <!--bclogo-->
```
This is an inaccurate role here. This is a link, not an action to submit a form or otherwise perform a the type of task a button would perform.

andrewpking replied:
Removed this role

taija commented on header.php:
```diff
@@ -126,7 +126,7 @@
 							wp_nav_menu( array( 
 								'menu' => 'main-nav', 
-								'items_wrap'      => '<ul id="%1$s" class="%2$s" role="navigation">%3$s</ul>',
+								'items_wrap'      => '<nav><ul id="%1$s" class="%2$s" role="navigation">%3$s</ul></nav>',
```
See notes earlier on labelling navs

andrewpking replied:
Now has a label

taija commented on header.php:
```diff
@@ -162,9 +162,9 @@
                                		 <form id="search" method="get" action="<?php echo esc_url( home_url( '/' ) ); ?>"> 
                                         <span aria-hidden="true" data-icon="&#xf002;"></span>
-                                        <input class="span3" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
+                                        <input role="search" class="span3" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>"/>
```
See notes on other pull request around incorrect usage of role

andrewpking replied:
Hoisted role=search up to form.

taija commented on header.php:
```diff
@@ -162,9 +162,9 @@
+                                        <input role="search" class="span3" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>"/>
 										<input type='hidden' name='post_type' value='programs,segments,staff,events,ads' />
-                                         <input id="searchsubmit" value="Search" type="submit" class="btn" />
+                                         <input role="button" id="searchsubmit" value="Search" type="submit" class="btn" />
```
See notes on other pull request around incorrect usage of role

andrewpking replied:
Removed this in commit.

taija commented on header.php:
```diff
@@ -181,7 +181,7 @@
 												wp_nav_menu( array( 
 													'menu' => 'main-nav', 
-													'items_wrap'      => '<ul id="%1$s" class="%2$s" role="navigation">%3$s</ul>',
+													'items_wrap'      => '<nav><ul id="%1$s" class="%2$s" role="navigation">%3$s</ul></nav>',
```
See notes earlier on labelling navs

andrewpking replied:
Now has aria-labelledby.

taija commented on searchform.php:
```diff
@@ -1,8 +1,8 @@
   <div class="input-append">
 		<span aria-hidden="true" data-icon="&#xf002;"></span>
-		<input class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
+		<input role="search" label="Search" class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>"/>
```
See notes on other pull request around incorrect usage of role

andrewpking replied:
resolved in commit.

taija commented on searchform.php:
```diff
@@ -1,8 +1,8 @@
+		<input role="search" label="Search" class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>"/>
 		<input type='hidden' name='post_type' value='programs,segments,staff,events,ads' />
-		<button id="searchsubmit" value="Search" type="submit" class="btn">Search</button>
+		<button label="button" id="searchsubmit" value="Search" type="submit" class="btn">Search</button>
```
See notes on other pull request around incorrect usage of role

andrewpking replied:
resolved in commit

taija commented on sidebar.php:
```diff
@@ -26,17 +26,17 @@
-	<div id="social-links">
-		<div>
+	<nav id="social-links">
+		<div role ="button">
```
This is incorrect usage of this role. First, if it had this role, it should be on the `<a>` tag, not on the `<div>`, since the div is not interactive. Second, these are semantically links not buttons.

andrewpking replied:
Fixed in commit

taija commented on sidebar.php:
```diff
@@ -26,17 +26,17 @@
 			<a class="btn btn-primary btn-block" href="https://www.facebook.com/KBCSBellevueSeattle"><i class="icon-facebook"></i> Facebook</a>
 		</div>
-		<div>
+		<div role="button">
```
See above comment regarding role

andrewpking replied:
fixed in commit

taija commented on sidebar.php:
```diff
@@ -26,17 +26,17 @@
 			<a class="btn btn-info btn-block" href="https://twitter.com/KBCS"><i class="icon-twitter"></i> Twitter</a>
 		</div>
-		<div>
+		<div role="button">
```
See above comment regarding role

andrewpking replied:
fixed in commit

taija commented on sidebar.php:
```diff
@@ -26,17 +26,17 @@
 
-	<div id="social-links">
-		<div>
+	<nav id="social-links">
```
See earlier comments regarding labelling navs

andrewpking replied:
Fixed in commit.

andrewpking committed:
Update footer.php

### Implemented Requested Changes

- No more nested navs
- Nav now has an aria-label
- Removed button label from BC image.

andrewpking committed:
Update header.php

### Implemented changes that were suggested

- Resolved improper use of aria-roles for buttons
- Resolved improper use of aria-roles for inputs
- Hoisted aria-role input to form element

andrewpking committed:
Update searchform.php

### Implemented changes requested

- Removed redundant role from button.
- Hoisted search role from input to form.

andrewpking committed:
Update sidebar.php

### Implemented changes to links nav

- Nav now has aria-labelledby
- Button roles were removed from links

andrewpking reviewed (commented):
I fixed all of the issues in commits to this branch. Please review when you have time @taija

taija commented on footer.php:
```diff
@@ -2,20 +2,18 @@
-		<div class="container" id="foot">
-			<div class="row">
+		<footer class="container" id="foot">
+			<nav aria-labelledby="Connect with KBCS" class="row">
```
Close but not quite there! aria-labelledby should include the ID of an element that labels the `<nav>`. If you would like to provide a label directly, using `aria-label` is the way to go instead.

## Example of aria-labelledby:
```html
<nav aria-labelledby="nav-heading">
    <h2 id="nav-heading">Related Programs</h2>
    ...
</nav>
```

## Example of aria-label
```html
<nav aria-label="Related Programs">
    ...
</nav>
```

andrewpking replied:
Thanks for the detailed explanation, I have resolved this issue using aria-label instead.

taija commented on footer.php:
```diff
@@ -64,16 +62,16 @@
 	
 	
-				<div class="span3">
+				<nav class="span3 bc-footer">
```
If you are using more than one `<nav>` on a given page, **all** navs need to be labelled.

andrewpking replied:
I meant to delete that nav, thanks for the catch. Deleted in a commit.

taija commented on header.php:
```diff
@@ -126,7 +126,7 @@
 							wp_nav_menu( array( 
 								'menu' => 'main-nav', 
-								'items_wrap'      => '<ul id="%1$s" class="%2$s" role="navigation">%3$s</ul>',
+								'items_wrap'      => '<nav aria-labelledby="main-nav"><ul id="%1$s" class="%2$s" role="navigation">%3$s</ul></nav>',
```
This is also incorrect use of aria-labelledby - `#main-nav` is not an element that provides a label.

andrewpking replied:
Fixed in a commit

taija commented on header.php:
```diff
@@ -181,7 +181,7 @@
 												wp_nav_menu( array( 
 													'menu' => 'main-nav', 
-													'items_wrap'      => '<ul id="%1$s" class="%2$s" role="navigation">%3$s</ul>',
+													'items_wrap'      => '<nav aria-labelledby="main-nav"><ul id="%1$s" class="%2$s" role="navigation">%3$s</ul></nav>',
```
See earlier comment on aria-labelledby

andrewpking replied:
Fixed in a commit

taija commented on searchform.php:
```diff
@@ -1,8 +1,8 @@
   <div class="input-append">
 		<span aria-hidden="true" data-icon="&#xf002;"></span>
-		<input class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>">
+		<input label="Search" class="input-xlarge" type="text" name="s" value="<?php echo trim( get_search_query() ); ?>"/>
```
This is not the correct use of label; label is not an attribute on the `<input>` tag. Either use `aria-label`, or add an element to serve as the label and use standard `<label for`> syntax

andrewpking replied:
Fixed in a commit

taija commented on sidebar.php:
```diff
@@ -26,7 +26,7 @@
 	</div>
 
-	<div id="social-links">
+	<nav aria-labelledby="social-links" id="social-links">
```
An element can not be labelled by itself.

andrewpking replied:
Thanks for the catch, I have changed this to aria-label in a commit.

taija reviewed (changes requested):
Several issues - please review and resolve. Thanks

andrewpking committed:
Update footer.php

### aria-labelledby replaced with aria-label

- Learned that labelledby points to an element by id
- Now the footer nav has a label that follows the semantics of the webpage.

andrewpking committed:
Update header.php

### Misunderstood use of aria-labelledby

- Now using aria-label's for nav elements
- Chose the label "Main" since these navs are the top level navigations for KBCS

andrewpking committed:
Update searchform.php

### Hoisted label to form element

- Fixed incorrect usage of label on input element
- Form element now has label of "Search"

andrewpking committed:
Update sidebar.php

### Resolved incorrect use of aria-labelledby

- Now using aria-label for social nav
- Changed label to "Social Links"

andrewpking reviewed (commented):
### Fixed issues with aria-labelledby and incorrect use of labels

- Resolved incorrect usage of label on an input element and hoisted to form element
- Resolved incorrect usage of aria-labelledby and replaced them with aria-labels
- Ensured the names for aria-labels are appropriate for content
